cardano-cli/src/Cardano/CLI/Compatible/StakeAddress/Run.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.CLI.Compatible.StakeAddress.Run
+  ( runCompatibleStakeAddressCmds
+  )
+where
+
+import Cardano.Api
+import Cardano.Api.Ledger qualified as L
+import Cardano.Api.Shelley
+
+import Cardano.CLI.Compatible.Exception
+import Cardano.CLI.Compatible.StakeAddress.Command
+import Cardano.CLI.Read
+import Cardano.CLI.Type.Error.StakeAddressRegistrationError
+import Cardano.CLI.Type.Key
+
+runCompatibleStakeAddressCmds
+  :: ()
+  => CompatibleStakeAddressCmds era
+  -> CIO e ()
+runCompatibleStakeAddressCmds = \case
+  CompatibleStakeAddressRegistrationCertificateCmd sbe stakeIdentifier mDeposit outputFp ->
+    runStakeAddressRegistrationCertificateCmd sbe stakeIdentifier mDeposit outputFp
+  CompatibleStakeAddressStakeDelegationCertificateCmd
+    sbe
+    stakeIdentifier
+    stkPoolVerKeyHashOrFp
+    outputFp ->
+      runStakeAddressStakeDelegationCertificateCmd sbe stakeIdentifier stkPoolVerKeyHashOrFp outputFp
+
+runStakeAddressRegistrationCertificateCmd
+  :: ()
+  => ShelleyBasedEra era
+  -> StakeIdentifier
+  -> Maybe Lovelace
+  -- ^ Deposit required in conway era
+  -> File () Out
+  -> CIO e ()
+runStakeAddressRegistrationCertificateCmd sbe stakeIdentifier mDeposit oFp = do
+  stakeCred <-
+    fromExceptTCli $
+      getStakeCredentialFromIdentifier stakeIdentifier
+
+  req <- createRegistrationCertRequirements sbe stakeCred mDeposit
+
+  let regCert = makeStakeAddressRegistrationCertificate req
+
+  fromEitherIOCli @_ @(FileError ()) $
```

Comment:
Interesting. We are prevented from specifying `@(CIO e)` because it is a type synonym. However `@(RIO e)` works.